src/main/kotlin/org/jetbrains/research/kotlin/mpp/inference/graph/Graph.kt

```diff
@@ -43,9 +43,25 @@ class Graph(
         return setInput(name, value)
     }
 
-    fun fetchOutputs(): List<Tensor<*>?> {
-        val out = nodes.filter { it.type == Node.NodeType.OUTPUT }
-        return out.flatMap { it.outputs.tensors.toList() }
+    inline fun <reified T : Number> setInput(tensor: Tensor<T>): Graph {
+        val name = tensor.name!!
+
+        require(name in availableInputs) { "Required input node is either already set or not found" }
+
+        nodes.find { name in it.inputs.names }!!.inputs[name] = tensor
+        return this
+    }
+
+    fun fetchOutputs(): List<Tensor<*>> {
+        val outputNodes = if (nodes.size == 1) nodes else nodes.filter { it.type == Node.NodeType.OUTPUT }
```

Comment:
`nodes.singleOrNull() ?: nodes.filter {...}`

Reply:
singleOrNull returns one element, but we need to work with the list in the future, so it's easier to just take the same list